tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,345 @@
+
+    def get_stack_outputs(self):
+        if not self.stack_description:
+            return {}
+        else:
+            output_key_to_value = {}
+            output_list = self.stack_description["Stacks"][0]["Outputs"]
+            for output in output_list:
+                output_key_to_value[output["OutputKey"]] = output["OutputValue"]
+            return output_key_to_value
+
+    def get_resource_status_by_logical_id(self, logical_id):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["LogicalResourceId"] == logical_id:
+                return res["ResourceStatus"]
+
+        return ""
+
+    def get_physical_id_by_type(self, resource_type):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["ResourceType"] == resource_type:
+                return res["PhysicalResourceId"]
+
+        return ""
+
+    def get_logical_id_by_type(self, resource_type):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["ResourceType"] == resource_type:
+                return res["LogicalResourceId"]
+
+        return ""
+
+    def get_physical_id_by_logical_id(self, logical_id):
+        if not self.stack_resources:
+            return ""
+
+        for res in self.stack_resources["StackResourceSummaries"]:
+            if res["LogicalResourceId"] == logical_id:
+                return res["PhysicalResourceId"]
+
+        return ""
+
+    def _fill_template(self, file_name):
+        """
+        Replaces the template variables with their value
+
+        Parameters
+        ----------
+        file_name : string
+            Template file name
+        """
+        input_file_path = str(Path(self.template_dir, file_name + ".yaml"))
+        updated_template_path = str(Path(self.output_dir, "sub_" + file_name + ".yaml"))
+        with open(input_file_path, "r") as f:
+            data = f.read()
+        for key, _ in CODE_KEY_TO_FILE_MAP.items():
+            data = data.replace("${{{}}}".format(key), self.get_code_key_s3_uri(key))
+        yaml_doc = yaml.load(data, Loader=yaml.FullLoader)
+
+        with open(updated_template_path, "w") as f:
+            yaml.dump(yaml_doc, f)
+
+        self.sub_input_file_path = updated_template_path
+
+    def set_template_resource_property(self, resource_name, property_name, value):
+        """
+        Updates a resource property of the current SAM template
+
+        Parameters
+        ----------
+        resource_name: string
+            resource name
+        property_name: string
+            property name
+        value
+            value
+        """
+        with open(self.sub_input_file_path, "r") as f:
+            data = f.read()
+        yaml_doc = yaml.load(data, Loader=yaml.FullLoader)
```

Comment:
Can we refactor this away? Reading a YAML is a common use case and we're repeating it.

Reply:
Added functions to load/write yaml